src/snowflake/connector/pandas_tools.py

```diff
@@ -219,12 +219,16 @@ def write_pandas(
     # see (https://docs.snowflake.com/en/user-guide/script-data-load-transform-parquet.html)
     if quote_identifiers:
         quote = '"'
-        columns = '"' + '","'.join(list(df.columns)) + '"'
-        parquet_columns = "$1:" + ",$1:".join(f'"{c}"' for c in df.columns)
+        # if the column name contains a double quote, we need to escape it by replacing with two double quotes
+        # https://docs.snowflake.com/en/sql-reference/identifiers-syntax#double-quoted-identifiers
+        snowflake_column_names = [str(c).replace('"', '""') for c in df.columns]
+        columns = '"' + '","'.join(snowflake_column_names) + '"'
+        parquet_columns = "$1:" + ",$1:".join(f'"{c}"' for c in snowflake_column_names)
     else:
         quote = ""
-        columns = ",".join(list(df.columns))
-        parquet_columns = "$1:" + ",$1:".join(df.columns)
+        snowflake_column_names = list(df.columns)
+        columns = ",".join(snowflake_column_names)
+        parquet_columns = "$1:" + ",$1:".join(snowflake_column_names)
```

Comment:
It might be an overkill, but I wonder if we could rewrite it to:

```
if quote_identifiers:
  quote = '"'
  snowflake_column_names = [str(c).replace('"', '""') for c in df.columns]
else:
  quote = ""
  snowflake_column_names = list(df.columns)

columns = quote + f"{quote},{quote}".join(snowflake_column_names) + quote
parquet_columns = "$1:" + ",$1:".join(f"{quote}{c}{quote}" for c in snowflake_column_names)
```

Reply:
I would prefer to keep the current one since it's more clear about what we do on `quote_identifiers`